### PR TITLE
[TensorRT EP] reduce CI pipelines test execution time

### PR DESCRIPTION
**Pull request by jywu-msft**
Currently the TensorRT EP CI test pipelines (running TensorRT 8.2) take around 1 hour 40 min to run.
This can be reduced by ~1 hour on Linux (40 min on Windows) with this change. For the upcoming TensorRT 8.4, the results are even more significant. (2.5 hours reduced to ~40 min)

The long test times is caused by TensorRT builder time initialization. ORT tests are run sequentially and builder instances are created and destroyed after each test. Nvidia advised adding a global unused builder object to keep the TRT kernel library in memory to avoid builder initialization/destruction costs.

### CODE REVIEW

File: onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc [hunk 1]
```diff
@@ -248,6 +248,12 @@ std::unique_lock<OrtMutex> TensorrtExecutionProvider::GetApiLock() const {
   return std::unique_lock<OrtMutex>(singleton);
 }
 
+#ifdef ORT_RUN_EXTERNAL_ONNX_TESTS
```
Comment (chilo-ms) [inline onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:251]: Just wondering where this MACRO will be defined. Is it defined in build script or cmake file?

Comment (jywu-msft) [reply onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:251]: in cmake.  when you set run_onnx_tests option in build script the cmake option gets enabled and that macro gets defined.

Comment (jywu-msft) [reply onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:251]: see https://github.com/microsoft/onnxruntime/blob/963e1ace4ed21fd7e368a97b077fc474ff551022/cmake/CMakeLists.txt#L1534

Comment (chilo-ms) [reply onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:251]: Okay, I see.

**Review by chilo-ms — DISMISSED** [review]

File: onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc [hunk 2]
```diff
@@ -248,6 +248,12 @@ std::unique_lock<OrtMutex> TensorrtExecutionProvider::GetApiLock() const {
...
+// instantiate global unused builder object which keeps the TRT kernel library in memory
+// so that subsequent builders avoid the expensive load / unload process.
+auto const placeholder = tensorrt_ptr::unique_pointer<nvinfer1::IBuilder>(nvinfer1::createInferBuilder(GetTensorrtLogger()));
```
Comment (edgchen1) [inline onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:254]: if it's only relevant for tests, would it be possible to put this global in a test source file? can the variable name be more specific? and perhaps make it static/in the unnamed namespace?

Comment (jywu-msft) [reply onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:254]: no, we only want TensorRT EP to depend on TensorRT libs, and not the OnnxRuntime test executable.

Comment (jywu-msft) [reply onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:254]: can make the variable ort-trt specific. and i can put it in an anonymous namespace or make it static.

**Review by edgchen1 — APPROVED** [review]

**Pull request merged**